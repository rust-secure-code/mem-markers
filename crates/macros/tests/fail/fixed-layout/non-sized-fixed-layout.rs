fn main() {
    fn ensure<T: mem_markers::FixedLayout>() {}
    ensure::<&str>();
}
