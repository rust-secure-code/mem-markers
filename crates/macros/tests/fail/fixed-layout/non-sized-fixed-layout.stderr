error[E0277]: the size for values of type `str` cannot be known at compilation time
  --> $DIR/non-sized-fixed-layout.rs:2:5
   |
2  |     mem_markers::fixed_layout::ensure::<&str>();
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ doesn't have a size known at compile-time
   | 
  ::: $WORKSPACE/src/fixed_layout.rs:10:18
   |
10 | pub fn ensure<T: FixedLayout>() {}
   |                  ----------- required by this bound in `mem_markers::fixed_layout::ensure`
   |
   = help: the trait `std::marker::Sized` is not implemented for `str`
   = note: to learn more, visit <https://doc.rust-lang.org/book/ch19-04-advanced-types.html#dynamically-sized-types-and-the-sized-trait>
   = note: required because of the requirements on the impl of `mem_markers::fixed_layout::FixedLayout` for `&str`
